projects/packages/blocks/src/class-blocks.php

```diff
@@ -99,7 +115,53 @@ function () use ( $feature_name, $method_name ) {
 			}
 		}
 
-		return register_block_type( $slug, $args );
+		return register_block_type( isset( $block_type ) ? $block_type : $slug, $args );
+	}
+
+	/**
+	 * Read block metadata from a block.json file.
+	 *
+	 * @param string $filename The path to the block.json file or its directory.
+	 *
+	 * @return array The block metadata.
+	 */
+	public static function get_block_metadata_from_file( $filename ) {
+		$metadata = array();
+		$needle   = '/block.json';
+		$filename = $needle === substr( $filename, -strlen( $needle ) ) ? $filename : $filename . $needle;
+
+		if ( file_exists( $filename ) ) {
+			try {
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+				$metadata = json_decode( file_get_contents( $filename ), true );
```

Comment:
Could we rely on `wp_json_file_decode` here?

Reply:
Wasn't aware of that function, thanks!